pkg/analysis/passes/sponsorshiplink/sponsorshiplink.go

```diff
@@ -0,0 +1,54 @@
+package sponsorshiplink
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/archive"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/metadata"
+)
+
+var (
+	sponsorshiplink = &analysis.Rule{Name: "sponsorshiplink", Severity: analysis.Recommendation}
+	explanation     = "Consider to add a sponsorship link in your plugin.json file (Info.Links section), which will be shown on the plugin details page to allow users to support your work if they wish."
+	recommendation  = "You can include a sponsorship link if you want users to support your work"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "sponsorshiplink",
+	Run:      checkSponsorshiplink,
+	Requires: []*analysis.Analyzer{metadata.Analyzer, archive.Analyzer},
+	Rules:    []*analysis.Rule{sponsorshiplink},
+	ReadmeInfo: analysis.ReadmeInfo{
+		Name:        "Sponsorship Link",
+		Description: "Checks if a sponsorship link is specified in `plugin.json` that will be shown in the Grafana plugin catalog for users to support the plugin developer.",
+	},
+}
+
+func checkSponsorshiplink(pass *analysis.Pass) (interface{}, error) {
+	metadataBody, ok := pass.ResultOf[metadata.Analyzer].([]byte)
```

Comment:
do plugins also put `links` in nested plugins plugin.json? if so you should be using `nestedmetadata` and checking every plugin json. e.g. https://github.com/grafana/plugin-validator/blob/15a7d6eb85255221cd48cf9aeff7a9595ac6cdba/pkg/analysis/passes/logos/logos.go#L8

Reply:
I think it is not necessary to handle this case here. We add this rule temporary just to show developers that they can configure such a link

Reply:
if this is temporary, please open an issue with a due date to remove it